util/merkle/src/node.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use elastic_array::ElasticArray1024;
+use rlp::*;
+use hashdb::DBValue;
+use primitives::H256;
+
+use nibbleslice::NibbleSlice;
+
+
+#[derive(Eq, PartialEq, Debug)]
+pub enum Node<'a> {
+    Leaf(NibbleSlice<'a>, DBValue),
+
+    Branch(NibbleSlice<'a>, [Option<H256>; 16]),
+}
+
+impl<'a> Node<'a> {
+    /// Decode the `node_rlp` and return the Node.
+    pub fn decoded(node_rlp: &'a [u8]) -> Option<Self> {
```

Comment:
Implement `Decodable` and `Encodable` instead of implementing customized functions.

Reply:
you mean trait encoding and decoding in util/rlp/src/trait.rs?

Reply:
Yes. That's more idiomatic way for handling rlp encoding in our code.

Reply:
I'm not sure about that since `encoded` function takes additional parameter, `size`.

Reply:
In that case, we can have two functions, one without size argument and the other accepting additional parameter.
Former one would be implementation of `Encodable`.

Reply:
This might cause large change of design so I think it's better to keep this. Additionally, I split `encoded` into two separate function